File: python/riot_sdk/core/route_cost.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class RouteCost:
    """Path cost (mm) to a station (BC-ROUTE-001). costs_mm=-1 means unreachable."""

    costs_mm: int

    @property
    def is_reachable(self) -> bool:
        return self.costs_mm >= 0

File: csharp/RIoT.Sdk.Core/RouteCost.cs
namespace RIoT.Sdk.Core;

/// <summary>
/// Path cost (mm) from a vehicle to a station on a Map (BC-ROUTE-001).
/// CostsMm = -1 means unreachable; that is a domain result, not a business failure (ADR-sdk-0005).
/// </summary>
public sealed record RouteCost(long CostsMm)
{
    public bool IsReachable => CostsMm >= 0;
}
